File: code/stable/glassbr/src/csharp/Control.cs
/** \file Control.cs
    \author Nikitha Krithnan and W. Spencer Smith
    \brief Controls the flow of the program
*/
using System;
using System.IO;

public class Control {
    
    /** \brief Controls the flow of the program
        \param args List of command-line arguments
    */
    public static void Main(string[] args) {
        StreamWriter outfile;
        string filename = args[0];
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'filename' assigned ");
        outfile.Write(filename);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        InputParameters inParams = new InputParameters();
        InputFormat.get_input(filename, inParams);
        DerivedValues.derived_values(inParams);
        InputConstraints.input_constraints(inParams);
        double h = Calculations.func_h(inParams);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'h' assigned ");
        outfile.Write(h);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        int GTF = Calculations.func_GTF(inParams);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'GTF' assigned ");
        outfile.Write(GTF);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        double AR = Calculations.func_AR(inParams);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'AR' assigned ");
        outfile.Write(AR);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        double q = Calculations.func_q(inParams);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'q' assigned ");
        outfile.Write(q);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        double q_hat = Calculations.func_q_hat(inParams, q, h, GTF);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'q_hat' assigned ");
        outfile.Write(q_hat);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        double J_tol = Calculations.func_J_tol(inParams, h);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'J_tol' assigned ");
        outfile.Write(J_tol);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        double J = Calculations.func_J(AR, q_hat);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'J' assigned ");
        outfile.Write(J);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        double q_hat_tol = Calculations.func_q_hat_tol(AR, J_tol);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'q_hat_tol' assigned ");
        outfile.Write(q_hat_tol);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        double B = Calculations.func_B(inParams, h, J);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'B' assigned ");
        outfile.Write(B);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        double NFL = Calculations.func_NFL(inParams, q_hat_tol, h);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'NFL' assigned ");
        outfile.Write(NFL);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        double P_b = Calculations.func_P_b(B);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'P_b' assigned ");
        outfile.Write(P_b);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        double LR = Calculations.func_LR(NFL, GTF);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'LR' assigned ");
        outfile.Write(LR);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        Boolean isSafePb = Calculations.func_isSafePb(inParams, P_b);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'isSafePb' assigned ");
        outfile.Write(isSafePb);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        Boolean isSafeLR = Calculations.func_isSafeLR(LR, q);
        outfile = new StreamWriter("log.txt", true);
        outfile.Write("var 'isSafeLR' assigned ");
        outfile.Write(isSafeLR);
        outfile.WriteLine(" in module Control");
        outfile.Close();
        OutputFormat.write_output(isSafePb, isSafeLR, P_b, J);
    }
}

File: code/stable/GlassBR/src/cpp/Control.cpp
/** \file Control.cpp
    \author Nikitha Krithnan and W. Spencer Smith
    \brief Controls the flow of the program
*/
#include <fstream>
#include <iostream>
#include <string>

#include "Calculations.hpp"
#include "DerivedValues.hpp"
#include "InputConstraints.hpp"
#include "InputFormat.hpp"
#include "InputParameters.hpp"
#include "OutputFormat.hpp"

using std::ofstream;
using std::string;

/** \brief Controls the flow of the program
    \param argc Number of command-line arguments
    \param argv List of command-line arguments
    \return exit code
*/
int main(int argc, const char *argv[]) {
    ofstream outfile;
    string filename = argv[1];
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'filename' assigned ";
    outfile << filename;
    outfile << " in module Control" << std::endl;
    outfile.close();
    InputParameters inParams = InputParameters();
    get_input(filename, inParams);
    derived_values(inParams);
    input_constraints(inParams);
    double J_tol = func_J_tol(inParams);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'J_tol' assigned ";
    outfile << J_tol;
    outfile << " in module Control" << std::endl;
    outfile.close();
    double q = func_q(inParams);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'q' assigned ";
    outfile << q;
    outfile << " in module Control" << std::endl;
    outfile.close();
    double q_hat = func_q_hat(inParams, q);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'q_hat' assigned ";
    outfile << q_hat;
    outfile << " in module Control" << std::endl;
    outfile.close();
    double q_hat_tol = func_q_hat_tol(inParams, J_tol);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'q_hat_tol' assigned ";
    outfile << q_hat_tol;
    outfile << " in module Control" << std::endl;
    outfile.close();
    double J = func_J(inParams, q_hat);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'J' assigned ";
    outfile << J;
    outfile << " in module Control" << std::endl;
    outfile.close();
    double NFL = func_NFL(inParams, q_hat_tol);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'NFL' assigned ";
    outfile << NFL;
    outfile << " in module Control" << std::endl;
    outfile.close();
    double B = func_B(inParams, J);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'B' assigned ";
    outfile << B;
    outfile << " in module Control" << std::endl;
    outfile.close();
    double LR = func_LR(inParams, NFL);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'LR' assigned ";
    outfile << LR;
    outfile << " in module Control" << std::endl;
    outfile.close();
    bool is_safeLR = func_is_safeLR(LR, q);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'is_safeLR' assigned ";
    outfile << is_safeLR;
    outfile << " in module Control" << std::endl;
    outfile.close();
    double P_b = func_P_b(B);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'P_b' assigned ";
    outfile << P_b;
    outfile << " in module Control" << std::endl;
    outfile.close();
    bool is_safePb = func_is_safePb(inParams, P_b);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'is_safePb' assigned ";
    outfile << is_safePb;
    outfile << " in module Control" << std::endl;
    outfile.close();
    write_output(is_safePb, is_safeLR, P_b, J);
    
    return 0;
}
